File: vlc-android/src/org/videolan/vlc/interfaces/IEventsHandler.java
package org.videolan.vlc.interfaces;

import android.support.v7.widget.RecyclerView;
import android.view.View;

import org.videolan.medialibrary.media.MediaLibraryItem;

public interface IEventsHandler {
    void onClick(View v, int position, MediaLibraryItem item);
    boolean onLongClick(View v, int position, MediaLibraryItem item);
    void onCtxClick(View v, int position, MediaLibraryItem item);
    void onUpdateFinished(RecyclerView.Adapter adapter);
}

File: vlc-android/src/org/videolan/vlc/interfaces/IEventsHandler.kt
package org.videolan.vlc.interfaces

import android.view.View
import androidx.recyclerview.widget.RecyclerView
import org.videolan.medialibrary.media.MediaLibraryItem

interface IEventsHandler {
    fun onClick(v: View, position: Int, item: MediaLibraryItem)
    fun onLongClick(v: View, position: Int, item: MediaLibraryItem): Boolean
    fun onImageClick(v: View, position: Int, item: MediaLibraryItem)
    fun onCtxClick(v: View, position: Int, item: MediaLibraryItem)
    fun onUpdateFinished(adapter: RecyclerView.Adapter<*>)
    fun onMainActionClick(v: View, position: Int, item: MediaLibraryItem)
}
